### PR TITLE
adding new feature isReviewed

### LINKED ISSUE #54: 🚀 - adding new isReviewed feature for products
# 🚀 feature request

**Is your feature request related to a problem? Please describe.**

added configuration files for isReviewed feature.  Files added were model, controller, and api route

**Describe the solution you'd like**

adding a feature to to review products where customers can add stars and comment

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by lucpizz**
## Proposed changes

This feature allows users/customers to add a review of the product after they purchase it.

## PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/davidsaulrodriguez/shop-portal/blob/master/CONTRIBUTING.md) doc
- [x] I have read the [BRANCHING MODEL](https://github.com/davidsaulrodriguez/shop-portal/blob/master/BRANCHINGMODEL.md) doc
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## PR Type

What types of changes does your code introduce to **Shop Portal**?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code Style Update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (Please describe):

## Current behavior

Issue Number: #54 

## New behavior

Fixes #54 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Further comments

N/A

### CODE REVIEW

File: controllers/isReviewedController.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const db = require('../models/isReviewedModel');
+
+// Defining methods for the postsController
+module.exports = {
+  findAll: function (req, res) {
+    db.find(req.query)
```
Comment (davidsaulrodriguez) [inline controllers/isReviewedController.js]: **this is pretty a serious bug**

```javascript
db.find(req.query) // Not Compliant
```

User provided data such as URL parameters, POST body-content should always be considered untrusted and tainted. Applications performing NoSQL operations based on tainted data could be exploited in a way similar to SQL Injection, where an attacker could inject NoSQL objects to access sensitive information or compromise data integrity.

To fix this and have a _compliant solution_ the data should be sanitized like so:

**example**

```javascript
function (req, res) {
  let query = { user: req.query.user.toString(), city: req.query.city.toString() };

  db.collection("users")
    .find(query) // Compliant
    .toArray((err, docs) => { });
}
```

**Review by davidsaulrodriguez — CHANGES_REQUESTED** [review]
A bug was introduced. Please review the comment I dropped on the line of code that introduces this bug.

File: controllers/isReviewedController.js (hunk 1 above)
Comment (lucpizz) [reply controllers/isReviewedController.js]: I added a fix by setting the query to a "toString()" method.

`findAll: function (req, res) {
    db.find(req.query.toString())`

Comment (lucpizz) [reply controllers/isReviewedController.js]: implemented a fix 

  create: function (req, res) {
    let reviewData = new db({
      name: req.body.name,
      imageUrl: req.body.imageUrl,
      imageKey: req.body.imageKey,
      description: req.body.description,
      totalStars: req.body.totalStars,
      reviewBody: req.body.reviewBody,
      isActive: req.body.isActive,
      isReviewable: req.body.isReviewable,
      created: req.body.created,
      updated: req.body.updated,
    });

    db.create(reviewData)
      .then((dbModel) => res.json(dbModel))
      .catch((err) => res.status(422).json(err));
  },

**Review by lucpizz — COMMENTED** [review]
reviewed the files

**Review by davidsaulrodriguez — APPROVED** [review]
Let's start getting this thing merged!

**Review by Delph-Sunny — APPROVED** [review]
Ok

**Pull request merged**